pages/[user]/[type].tsx

```diff
@@ -233,72 +234,73 @@ export default function Type(props): Type {
   }
 }
 
-export const getServerSideProps: GetServerSideProps = async (context: GetServerSidePropsContext) => {
-  const dateQuery = context.query?.date ?? null;
-  const date = Array.isArray(dateQuery) && dateQuery.length > 0 ? dateQuery.pop() : dateQuery;
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  // get query params and typecast them to string
+  // (would be even better to assert them instead of typecasting)
+  const userParam = context.query.user as string;
+  const typeParam = context.query.type as string;
+  const dateParam = context.query.date as string | undefined;
```

Comment:
Kinda unrelated, but we can probably simplify this by into the page itself; as dateParam is actually unused within getServerSideProps and the client code has access through the router.

Reply:
💯  

I didn't do any actual changes to the code - just added types

Reply:
You might like this little hacked lib I made a while back -  https://github.com/KATT/next-use-query-params

Reply:
I actually do like that, that's pretty neat.